src/cudamatrix/cu-matrix.cc

```diff
@@ -2012,30 +2040,116 @@ void CuMatrixBase<Real>::AddCols(const CuMatrixBase<Real> &src,
 }
```

Comment:
I mean maybe the version of AddCols that take std::vector as index list is not necessary any more. Because I notice that the version of CopyCols that take std::vector as index list is deleted.